go/coding_standard.md

```diff
@@ -31,18 +31,29 @@ Please add items here, or modify them as appropriate. We have git versioning, do
  * Many comments are not sentences, these should begin with a lower case letter and end without a period.
  * The first letter of sentences in comments are capitalized and ends with a period.
  
-## Git
-
- * Production repos never push to `master` or `develop` branches except the push-master.
- * `develop` should always pass tests.
- * Always start work with a "WIP: \*" (work-in-progress) pull-request, on a branch ideally named "feature/\*" or "hotfix/\*". You're encouraged to fork the repo and add it as a remote in .git/config (so you can push to your fork, but still use the same import paths to the original project).
- * When complete, pull requests should be opened to the `develop` branch.
- * Do not merge PRs that aren't final -- e.g. no code dead code or code that won't be used should be committed.
- * Squash and merge to keep the history clean.
- * Merging from `develop` to `master` happens during a release.
- * All branches should be in lowercase, underscores are okay in branch names
- * All pull requests should include any relevant additions to the `CHANGELOG.md`, If PR is to the develop branch changes
- can be included under an `unreleased changes` header at the top of the document.
+## Github
+
+ * Production branches
+   * Production repos never push to `master` or `develop` branches except the push-master.
+   * `develop` should always pass tests.
+   * `develop` branch should have an on going open PR to `master`, high level comments can be made here
+   * Merging from `develop` to `master` happens during a release.
+ * Feature branches
+   * should following the naming convention `yourname/branch-description`
+   * all branches should be in lowercase, underscores/dashes are okay in branch names
+   * should have an open PR with the name "WIP: your feature description"
+   * All feature PRs should opened to merge to the `develop` branch, unless stacked on other PR leading to `develop`.
+    * Ideally, for faster merges, break your PR into several PRs that stack on top of each other and lead to `develop`.
+ * Reference branches
+   * Sometimes is is necessary to keep a branch with reference information even there there is no intention of merging 
+   * These branches should be named `yourname/ref/branch-description`
+ * Pull Requests (PRs)
+   * Do not merge PRs that aren't final -- e.g. no code dead code or code that won't be used should be committed.
+   * Always delete branch immediately after merge
+   * [Squash](https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git) 
+   commits to keep the history clean.
+   * All pull requests should include any relevant additions to the `CHANGELOG.md`, 
+   If PR is to the develop branch changes can be included under an `unreleased changes` header at the top of the document.
```

Comment:
Just for clarity: "If the PR is opened against the develop branch, then include the changes under an `unreleased changes` header at the top of the `CHANGELOG.md` document."

Reply:
The released changes will be in the `CHANGELOG.md` on master right? I think we should add to `CHANGELOG.md` as if this is master (so every PR could be part of the release without trouble). I think having a `unreleased changes` section will cause more work/trouble.

Reply:
This seems overly complicated and will lead to missing additions. Can't we compile the changelog additions from the delta since last merge to master? As we expect only completed PRs to be merged in and not single commits, this should make for a clear overview of what has changed. Down to that it's possible to do it in an automatic fashion.

Reply:
What would you base the delta on? PRs often hold more then one change. And parsing all commit messages sounds like information overload. But I like the idea of automation. Maybe the PR has to state exactly what was done in a formatted way. This way it is also easier to read for approvers.

Reply:
Indeed, I assume that both the title of the PR as well as the description communicate clearly what to expect from the change. The changelog should be a high-level index for the curious to get an overview and pointers where to look for more info, an example line:

* Concise description ([Link to PR](http://very-nice-pr.org))

Check the Hashicorp projects for inspiration: [Terraform Changelog](https://github.com/hashicorp/terraform/blob/master/CHANGELOG.md).

Reply:
@xla nice! I like the Terraform changelog - I'm adamant that it is a good idea to track unreleased changes as done there.

Reply:
> Check the Hashicorp projects for inspiration: Terraform Changelog.

I think I copied their changelog style verbatim.

Automating this sounds hard, we'd have to be super disciplined in the PR descriptions. Maybe it's a long term ideal. In the meantime I suppose it should be ok to just get it from the set of merged PRs since the last release. Ideally folks would update the changelog [unreleased] as they go to simplify this but I'm worried about the merge conflicts